automated_packaging/prepare_release.pl

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/perl
+use lib '/usr/local/bin';
+use common_functions;
+use JSON;
+
+$NEW_VERSION = $ARGV[0];
+$PROJECT = $ARGV[1];
+$ = $ARGV[2];
+
+# Cherry-pick commit using commit end-points
+sub cherry_pick_commits {
+	@commit_urls = @_;
+
+	my $minor_version = substr($NEW_VERSION, 0, 3);
+	`git checkout release-$minor_version`;
+
+	foreach my $commit_url ( @commit_urls ) {
+		my $current_commits_text = `curl -H "Accept: application/vnd.github.v3.full+json" -X GET --user "$github_token:x-oauth-basic" '$commit_url'`;
+		my @current_commits = @{decode_json($current_commits_text)};
+
+		foreach my $current_commit ( @current_commits ) {
+			%current_commit_hash = %$current_commit;
+
+			# Cherry pick commits
+			my $current_sha_code = $current_commit_hash{'sha'};
+			`git cherry-pick $current_sha_code`;
+		}
+	}
+}
+
+my $github_token = get_and_verify_token();
+my $minor_version = substr($NEW_VERSION, 0, 3);
+
+# We will update the new major-minor version
+$minor_version_escape_dot = $minor_version;
+$minor_version_escape_dot =~ s/[^0-9]/\\./g;
```

Comment:
Let's use `s/\./\\./g`. In current version, it is difficult to understand we escape dots.